internal/controllers/reference/reference.go

```diff
@@ -115,6 +155,64 @@ func removeOutdatedReferencesToSecret(
 	return nil
 }
 
+// removeOutdatedReferencesToConfigMap removes outdated reference records to configMaps in reference indexer.
+// ConfigMaps that are referred by referrer are passed in referredConfigMapNameMap parameter.
+// If a ConfigMap is not referenced by any other object after deleting outdated reference records,
+// and it does not have label "konghq.com/ca-cert:true", it is not possible to be used in Kong gateway config
+// and should be removed from the object cache inside KongClient.
+func removeOutdatedReferencesToConfigMap(
```

Comment:
This looks very similar to `removeOutdatedReferencesToSecret`. Is there a chance we could refactor those to deduplicate the code?

Reply:
Done in de81f12fd9a7aa3bf513b2e2db26b2d7de826eec